pkg/cri/server/image_pull.go

```diff
@@ -266,26 +270,36 @@ func ParseAuth(auth *runtime.AuthConfig, host string) (string, string, error) {
 // Note that because create and update are not finished in one transaction, there could be race. E.g.
 // the image reference is deleted by someone else after create returns already exists, but before update
 // happens.
-func (c *criService) createImageReference(ctx context.Context, name string, desc imagespec.Descriptor) error {
+func (c *criService) createImageReference(ctx context.Context, name string, desc imagespec.Descriptor, labels map[string]string) error {
 	img := containerdimages.Image{
 		Name:   name,
 		Target: desc,
 		// Add a label to indicate that the image is managed by the cri plugin.
-		Labels: map[string]string{imageLabelKey: imageLabelValue},
+		Labels: labels,
 	}
 	// TODO(random-liu): Figure out which is the more performant sequence create then update or
 	// update then create.
 	oldImg, err := c.client.ImageService().Create(ctx, img)
 	if err == nil || !errdefs.IsAlreadyExists(err) {
 		return err
 	}
-	if oldImg.Target.Digest == img.Target.Digest && oldImg.Labels[imageLabelKey] == imageLabelValue {
+	if oldImg.Target.Digest == img.Target.Digest && oldImg.Labels[crilabels.ImageLabelKey] == crilabels.ImageLabelValue {
```

Comment:
should this be 

```
... && oldImg.Labels[crilabels.ImageLabelKey] == labels[crilabels.ImageLabelKey]
```

?

Reply:
not sure, `crilabels.ImageLabelKey` and `crilabels.ImageLabelValue` is constant,  so comparing against it, we don't know from where `labels` are coming (although here getLabels are called everytime before calling `creatingImageRef`) but it might not every time in future this function may used elsewhere and labels could be different.

Reply:
> we don't know from where labels are coming (although here getLabels are called everytime before calling creatingImageRef) but it might not every time in future this function may used elsewhere and labels could be different.

Exactly. If in future containerd has a new possible value for `crilabels.ImageLabelKey` (let's call it "new_managed"), and

1.  `createImageReference` is called with `labels{"io.cri-containerd.image":"new_managed"}`, 
2.  `oldImg` here has  `labels{"io.cri-containerd.image":"managed"}`

then the current CL will just return nil, right? But what we want to do is to update the value for `crilabels.ImageLabelKey`.